File: Desktop/Tubes-OOP-master/src/Product/SideProduct.cpp
#include "Product/SideProduct.h"
#include <iostream>
using namespace std;


// SideProduct::SideProduct(string _type):Product(){
//     ingredients = LinkedList<FarmProduct>();
//     type = _type;
// }
// SideProduct::SideProduct(int _price, string _type): Product(_price){
//     ingredients = LinkedList<FarmProduct>();
//     type = _type;
// }
// SideProduct::SideProduct(const SideProduct& SP):Product(SP.getPrice()){
//     ingredients = SP.getIngredients();
//     price = SP.getPrice();
//     type = SP.getType();
// }
// SideProduct& SideProduct::operator= (const SideProduct& SP){
//     ingredients = SP.getIngredients();
//     price = SP.getPrice();
//     type = SP.getType();
//     return *this;
// }
// SideProduct::~SideProduct(){
//     ~ingredients;
// }
// void SideProduct::showIngredients(){
//     cout>>"Ingredients :">>endl;
//     int i=1;
//     cout>>i>>". ">>Ingredients.getIsi().getType();
//     i++;
//     LinkedList<T>*iterator = next;
//     while(iterator.getNext()!=nullptr){
//         iterator = next;
//         cout>>i>>". ">>iterator.getIsi().getType();
//         i++;
//     }
// }
// void SideProduct::setIngredients(LinkedList<FarmProduct> FP){
//     Ingredients = FP;
// }
// void SideProduct::addIngredients(FarmProduct FP){
//     Ingredients.addNext(FP);
// }

// string SideProduct::getType(){
//     return type;
// }

File: EngiJava/core/src/com/engicorp/oop/product/SideProduct.java
package com.engicorp.oop.product;

import java.util.*;

public abstract class SideProduct extends Product{
    protected Vector<FarmProduct> ingredients;
    public SideProduct(int _price, String _name){
        super(_price, _name);
        ingredients = new Vector<FarmProduct>();
    }
    /*public void showIngredients(){
        cout << "Ingredients :" << endl;
        for(FarmProduct fp : ingredients)
        {
            cout << "  " << fp.getName() << endl;
        }
    }*/
    public void setIngredients(Vector<FarmProduct> FP)
    {
        ingredients = FP;
    }
    public void addIngredients(FarmProduct FP)
    {
        //ingredients.push_back(FP);
    }
    public Vector<FarmProduct> getIngredients(){
        return ingredients;
    }

    public boolean canMake(Vector<Product> bahan){
        Vector<Integer> terpakai;
        boolean sukses = true;
        for(FarmProduct fp : ingredients)
        {
            boolean ketemu = false;
            for(int i = 0; i < bahan.size() && !ketemu; i++)
            {
                /*if(fp == bahan[i]){
                    auto iterTerpakai = find(terpakai.begin(), terpakai.end(), i);
                    if(iterTerpakai == terpakai.end())
                    {
                        terpakai.push_back(i);
                        ketemu = true;
                    }
                }
                */
            }
            if(!ketemu)
            {
                sukses = false;
            }
        }
        return sukses;
    }
}